nservicebus/messaging/routing.md

```diff
@@ -1,172 +1,68 @@
-
-snippet:Routing-CustomRoutingStore
-
-NOTE: The function passed to the `SetFallbackRoute` call is executed **each time** a message is sent so it is essential to make sure it is performant (e.g. by caching the results if getting the result requires crossing a process boundary).
+## Routing concepts
 
+NServiceBus routing consists of two layers, logical and physical. Logical routing defines which logical endpoint should receive a given outgoing message. Physical routing defines to which actual physical instance of the selected endpoint should the message be delivered. While logical routing is a developer's concern, physical routing is controlled by operations. Apart from MSMQ transport physical routing usually is handled automatically by NServiceBus and does not require any configuration. To read more about physical routing with MSMQ go [here](/nservicebus/msmq/routing.md).
 
-### Endpoint instance mapping layer
 
-The purpose of the endpoint instance mapping layer is to provide information about physical deployments of a given endpoint. Endpoint instance mapping is optional for most transports in most cases because of built-in conventions. An example of situation where it is required is a scaled out endpoint using MSMQ transport. Other endpoints that send messages to it need to be aware of its instances.
+## Command routing
 
+As described [here](/nservicebus/messaging/messages-events-commands.md), NServiceBus distinguishes several types of messages. Command messages are always routed to a single destination. 
 
-#### Using config file
+partial:commands
 
-When using MSMQ transport, endpoint instance mapping can be done using the [file based instance mapping](/nservicebus/msmq/scalability-and-ha/sender-side-distribution.md).
+snippet:endpoint-mapping-appconfig
 
+The per-namespace routes override assembly-level routes and the per-type routes override both namespace and assembly routes.
 
-#### Static mapping
 
-The `EndpointInstances` class provides a method that allows the registration of a static mapping. This API is useful in specific scenarios where developers need to take control over a certain mapping.
+### Overriding the destination
 
-snippet:Routing-StaticEndpointMapping
+In specific cases the routing configuration can be overridden when sending a given message (e.g. send to a particular address or send to self). Refer to documentation on [sending](/nservicebus/messaging/send-a-message.md) for further details.
 
 
-#### Dynamic mapping
+## Event routing
 
-Dynamic mapping is meant to provide a convenient extension for NServiceBus add-ons. To add a dynamic mapping rule pass a function that takes an endpoint name and returns a collection of endpoint instance names.
+Published event messages are routed to a single instance of each subscriber endpoint. However, before the message is published and delivered, the subscriber has to express its interest in a given type of event. 
```

Comment:
> Published event messages are routed to a single instance of each subscriber endpoint

Events can be received by multiple logical endpoints, however in case of scale-out each event will be received only by one physical instance of any logical endpoint.